### PR TITLE
Template for adding task

### PR DESCRIPTION
**Pull request by maciejdrobnik**
I created the template for adding task. In this PR is also the adding the roles and I used it for addTask button which is visible only for team manager.

### CODE REVIEW

File: lib/components/sub_pages/calendar_details/add_task/calendar_details_add_task.dart [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+                            firstDate: DateTime(1950),
+                            lastDate: DateTime(2100));
+                        if (pickedStartDate != null) {
+                          TimeOfDay? pickedStartTime = await showTimePicker(
+                              context: context, initialTime: TimeOfDay.now());
+                          changeDate("start", pickedStartTime, pickedStartDate);
+                        }
+                      },
+                      child: Text(
+                          "Start: ${start.day} ${timeService.getMonthFromNumber(start.month)} ${start.hour}:${start.minute}")),
```
Comment (mirimkim00) [inline lib/components/sub_pages/calendar_details/add_task/calendar_details_add_task.dart:150]: Don't we need any hyphens between day and month?

File: lib/components/sub_pages/calendar_details/add_task/calendar_details_add_task.dart [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+  ConvertTimestampService timeService = ConvertTimestampService();
+  FirebaseService databaseService = FirebaseService();
+
+  @override
+  void initState() {
+    typeSelection = listOfTypes.first.value;
+    descriptionController = TextEditingController();
+    nameController = TextEditingController();
+    name = "";
+    description = " ";
```
Comment (mirimkim00) [inline lib/components/sub_pages/calendar_details/add_task/calendar_details_add_task.dart:42]: Do you initiate it as a space?

File: lib/components/sub_pages/calendar_details/add_task/calendar_details_add_task.dart [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+                            firstDate: DateTime(1950),
+                            lastDate: DateTime(2100));
+                        if (pickedEndDate != null) {
+                          TimeOfDay? pickedEndTime = await showTimePicker(
+                              context: context, initialTime: TimeOfDay.now());
+                          changeDate("end", pickedEndTime, pickedEndDate);
+                        }
+                      },
+                      child: Text(
+                          "End: ${end.day} ${timeService.getMonthFromNumber(end.month)} ${end.hour}:${end.minute}")),
```
Comment (mirimkim00) [inline lib/components/sub_pages/calendar_details/add_task/calendar_details_add_task.dart:165]: Don't we need any hyphens between day and month?

**Review by mirimkim00 — CHANGES_REQUESTED** [review]

File: lib/components/sub_pages/calendar_details/add_task/calendar_details_add_task.dart (hunk 2 above)
Comment (maciejdrobnik) [reply lib/components/sub_pages/calendar_details/add_task/calendar_details_add_task.dart:42]: It should be empty string probably

File: lib/components/sub_pages/calendar_details/calendar_details_content.dart [hunk 4]
```diff
@@ -119,19 +128,28 @@ class _CalendarDetailsContentState extends State<CalendarDetailsContent> {
...
-              )),
+          if (widget.data!.role == "teammanager")
+            SizedBox(
+                height: constants.calendarDetailsIconRowHeight *
+                    widget.parentConstraints.maxHeight,
+                child: Row(
+                  mainAxisAlignment: MainAxisAlignment.end,
+                  children: [
+                    Padding(
+                        padding: const EdgeInsets.only(right: 15),
```
Comment (mirimkim00) [inline lib/components/sub_pages/calendar_details/calendar_details_content.dart:139]: magic number

File: lib/components/sub_pages/calendar_details/calendar_details_role_check.dart [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        future: FirebaseService().getUserData(),
+        builder: (context, snapshot) {
+          if (snapshot.connectionState == ConnectionState.done) {
+            if (snapshot.hasData) {
+              return CalendarDetailsContent(
+                  events: events,
+                  parentConstraints: parentConstraints,
+                  data: snapshot.data);
+            } else {
+              return Text("Non oggi");
```
Comment (mirimkim00) [inline lib/components/sub_pages/calendar_details/calendar_details_role_check.dart:27]: Is it polish?

File: lib/components/sub_pages/calendar_details/calendar_details_role_check.dart [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+            }
+          } else {
+            return Text("Aspetta un momento perfavore");
```
Comment (mirimkim00) [inline lib/components/sub_pages/calendar_details/calendar_details_role_check.dart:30]: polish again...?

**Review by mirimkim00 — CHANGES_REQUESTED** [review]

**Review by mirimkim00 — APPROVED** [review]

**Pull request merged**